tools/stress-cluster/services/Stress.Watcher/src/JobEventHandler.cs

```diff
@@ -0,0 +1,237 @@
+                return;
+            }
+
+            var rgName = GetResourceGroupName(job);
+
+            if (string.IsNullOrEmpty(rgName))
+            {
+                return;
+            }
+
+            Subscription subscription = ARMClient.DefaultSubscription;
+
+            ResourceGroup resourceGroup;
+            try {
+                resourceGroup = await subscription.GetResourceGroups().GetAsync(rgName);
+            } catch (Exception) {
+                Logger.Error($"Failed to get resource group '{rgName}' using subsription id '{subscription.Id}'");
+                return;
+            }
+
+            Logger.Information($"Deleting resources for group {rgName}");
+            await resourceGroup.DeleteAsync();
+            Logger.Information($"Deleted resources for group {rgName}");
+        }
+
+        public bool ShouldDeleteResources(V1Job job)
+        {
+            if (!string.IsNullOrEmpty(Namespace) && Namespace != job.Namespace())
+            {
+                return false;
+            }
+
+            var initContainers = job.Spec?.Template?.Spec?.InitContainers;
+            if (initContainers == null || initContainers.Count() == 0) {
+                return false;
+            }
+
+            var deployContainers = initContainers.Where(c => c.Name == "init-azure-deployer");
+            if (deployContainers.Count() == 0)
+            {
+                return false;
+            }
+
+            /*
+               Check if job is done by determing whether any conditions are satifisfied as "True"
+               This is most flexible because checking succeeded/failed counts will not be flexible
+               when there are multiple pods specified for the job.
+
+               Example succeeded state
+               --------------------
+                 status:
+                   completionTime: "2022-10-26T20:53:20Z"
+                   conditions:
+                   - lastProbeTime: "2022-10-26T20:53:20Z"
+                     lastTransitionTime: "2022-10-26T20:53:20Z"
+                     status: "True"
+                     type: Complete
+                   ready: 0
+                   startTime: "2022-10-26T20:52:28Z"
+                   succeeded: 1
+               --------------------
+
+               Example failed state
+               --------------------
+                 status:
+                    conditions:
+                    - lastProbeTime: "2022-10-26T21:05:39Z"
+                      lastTransitionTime: "2022-10-26T21:05:39Z"
+                      message: Job has reached the specified backoff limit
+                      reason: BackoffLimitExceeded
+                      status: "True"
+                      type: Failed
+                    failed: 1
+                    ready: 0
+                    startTime: "2022-10-26T21:04:58Z"
+               --------------------
+           */
+            var isCompleted = job.Status?.Conditions?.Any(c => c.Status == "True");
```

Comment:
need to take into account isCompleted == null

Reply:
I'm doing the `isCompleted == true` checks to handle the null case.

Reply:
return false with debug message "Error getting job status"  ??